docs/server/upgrading.rst

```diff
@@ -3,170 +3,222 @@
+Clean up stale python bytecode
+------------------------------
 
-Upgrading
----------
+You should remove any stale python bytecode files before upgrading.
+
+Assuming you are in the root of your virtualenv folder you can run:
+
+.. code-block:: bash
+
+   (env) $ find . -name ".pyc" | xargs rm
 
-Upgrading Pootle using the :command:`pip`.
 
-.. note:: You will need to adjust these instructions if you installed Pootle
-   using another method, such as directly from a Git checkout.
+.. _upgrading#upgrading-2.6:
+
+
+Upgrading from a version prior to 2.6
+-------------------------------------
+
+If you are upgrading from a version prior to 2.6 you will need to first upgrade
+to the latest 2.6.x version and then you will be able to upgrade to the latest
+version.
+
+.. code-block:: bash
+
+   (env) $ pip install --upgrade "Pootle>=2.6,<2.7"
+   (env) $ pootle setup
 
 .. warning::
+   The 2.6.x releases are meant only as a migration step.
+
+   You should upgrade immediately to the latest version once setup has
+   completed.
+
+
+.. _upgrading#upgrading-latest:
+
+Upgrading from version 2.6.x
+----------------------------
+
+Upgrade to the latest Pootle package:
+
+.. code-block:: bash
+
+   (env) $ pip install --upgrade Pootle
+
+
+.. _upgrading#check-settings:
+
+Update and check your settings
+------------------------------
+
+You should now update your custom Pootle settings to add remove or adjust any
+settings that have changed. You may want to view the latest 
+:ref:`available settings <settings#available>` 
+
+You can check to see if there are any issues with your configuration
+settings that need to be resolved:
+
+.. code-block:: bash
+
+   (env) $ pootle check
+
+.. note:: If you are upgrading from a version of Pootle that uses
+   :file:`localsettings.py` then you must :ref:`move your custom settings
+   <settings#customizing>` to a new location in order to ensure that Pootle
+   uses them.
+
+.. _upgrading#schema-migration:
+
+Migrate your database schema
+----------------------------
+
+Once you have updated your settings you can perform the database schema and
+data upgrade by running:
+
+.. code-block:: bash
+
+   (env) $ pootle migrate
+
 
-   Always backup the following before upgrading:
+.. _upgrading#refresh-stats:
 
-   - the entire **database**
-   - all the **settings**
-   - all your **translation files**
-   - any **code customizations**
-   - any **templates customizations**
-   - any **styling customizations**
+Refreshing checks and stats
+---------------------------
 
+You should now update the translation checks and populate the Redis cache with
+statistical data.
 
-To perform the upgrade follow the next steps:
+If you have not already done so you should
+:ref:`install and start a Redis server <requirements#packages>`.
 
-* If you want to perform a :doc:`database migration <database_migration>` then
-  do it first.
+As we haven't started the RQ worker yet we will run the commands using the
+``--no-rq`` option:
```

Comment:
The thing is i think there are very few that have managed to upgrade to 2.7.0

I also get back to the point that this will produce far better information and is less likely to go wrong.